src/jdk.compiler/share/classes/com/sun/tools/javac/code/Lint.java

```diff
@@ -417,9 +417,22 @@ public enum LintCategory {
         }
 
         LintCategory(String option, boolean annotationSuppression) {
+            this(option, annotationSuppression, null);
+        }
+
+        LintCategory(String option, LintCategory alias) {
+            this(option, true, alias);
+        }
+
+        LintCategory(String option, boolean annotationSuppression, LintCategory alias) {
             this.option = option;
             this.annotationSuppression = annotationSuppression;
+            this.alias = alias;
             map.put(option, this);
+            // we need to do this as forward references are not allowed
+            if (alias != null) {
+                alias.alias = this;
```

Comment:
I think we wish to make this a union-find-like structure (if you are familiar with leetcode problems) - we need to have some category as roots that point to themselves. Other category like synchronization can point to identity. When we use a `LintCategory` we do a "find root" to find the actual category to use.